libraries/rtprocessing/icp.cpp

```diff
@@ -0,0 +1,288 @@
+// QT INCLUDES
+//=============================================================================================================
+
+#include <QSharedPointer>
+#include <QDebug>
+
+//=============================================================================================================
+// EIGEN INCLUDES
+//=============================================================================================================
+
+#include <Eigen/Core>
+#include <Eigen/Dense>
+#include <Eigen/Eigenvalues>
+#include <Eigen/Geometry>
+
+//=============================================================================================================
+// USED NAMESPACES
+//=============================================================================================================
+
+using namespace MNELIB;
+using namespace RTPROCESSINGLIB;
+using namespace Eigen;
+using namespace FIFFLIB;
+
+//=============================================================================================================
+// DEFINE GLOBAL METHODS
+//=============================================================================================================
+
+bool RTPROCESSINGLIB::icp(const MNEProjectToSurface::SPtr mneSurfacePoints,
+                          const Eigen::MatrixXf& matPointCloud,
+                          FiffCoordTrans& transFromTo,
+                          const int iMaxIter,
+                          const float fTol,
+                          const VectorXf& vecWeitgths)
+/**
+ * Follow notation of P.J. Besl and N.D. McKay, A Method for
+ * Registration of 3-D Shapes, IEEE Trans. Patt. Anal. Machine Intell., 14,
+ * 239 - 255, 1992.
+ */
+{
+    // Initialization
+    int iNP = matPointCloud.rows();             // The number of points
+    float fMSEPrev,fMSE = 0.0;                  // The mean square error
+    float fScale = 1.0;
+    float bScale = true;
+    MatrixXf matP0 = matPointCloud;             // Initial Set of points
+    MatrixXf matPk = matP0;                     // Transformed Set of points
+    MatrixXf matYk(matPk.rows(),matPk.cols());  // Iterative closest points on the surface
+    MatrixXf matDiff = matYk;
+    VectorXf vecSE(matDiff.rows());
+    Matrix4f matTrans;                          // the transformation matrix
+    VectorXi vecNearest;                        // Triangle of the new point
+    VectorXf vecDist;                           // The Distance between matX and matP
+
+    // Initial transformation - From point cloud To surface
+    FiffCoordTrans transICP = transFromTo;
+    matPk = transICP.apply_trans(matPk);
+
+    // Icp algorithm:
+    for(int iIter = 0; iIter < iMaxIter; ++iIter) {
+
+        // Step a: compute the closest point on the surface; eq 29
+        if(!mneSurfacePoints->mne_find_closest_on_surface(matPk, iNP, matYk, vecNearest, vecDist)) {
+            qWarning() << "RTPROCESSINGLIB::icp: mne_find_closest_on_surface was not sucessfull.";
```

Comment:
- This should read [RTPROCESSINGLIB::icp]. See docu [here](https://mne-cpp.github.io/pages/contribute/conv_style.html#command-line-outputs). change rest of you PR accordingly please.